### PR TITLE
Awais786/ecom 3280 cert url

### PR DESCRIPTION
**Pull request by awais786**
Add simple endpoint for viewing credentials by its uuid in credentials service
Add a simple render-view 
Update test cases.

### CODE REVIEW

**Comment by awais786** [comment]
@clintonb test cases failing on travis. Without showing any actual issue. It seems like it is compression issue.  any ideas ?

**Comment by clintonb** [comment]
@awais786 the first error message tells you everything you need to know to fix the problem. I'm going to challenge you to resolve this on your own. In the process you may learn a bit about how django-compressor works.

File: credentials/apps/api/serializers.py [hunk 1]
```diff
@@ -82,6 +99,7 @@ class UserCredentialCreationSerializer(serializers.ModelSerializer):
     """ Serializer used to create UserCredential objects. """
     credential = CredentialField()
     attributes = UserCredentialAttributeSerializer(many=True)
+    certificate_url = UserCertificateURLField(source='uuid')
```
Comment (clintonb) [inline credentials/apps/api/serializers.py]: No URL is needed at the time of creation.

File: credentials/apps/api/tests/test_views.py [hunk 2]
```diff
@@ -195,7 +198,10 @@ def test_create_with_programcertificate(self):
         response = self._attempt_create_user_credentials(data)
         self.assertEqual(response.status_code, 201)
         user_credential = UserCredential.objects.get(username=username)
-        self.assertEqual(json.loads(response.content), UserCredentialSerializer(user_credential).data)
+        self.assertEqual(
+            json.loads(response.content),
```
Comment (clintonb) [inline credentials/apps/api/tests/test_views.py]: response.data

File: credentials/apps/credentials/views.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+
+logger = logging.getLogger(__name__)
+
+
+class RenderCredential(TemplateView):
+    """ Certificate rendering view."""
+    template_name = 'credentials/view_certificate.html'
+
+    credit_api = None
```
Comment (clintonb) [inline credentials/apps/credentials/views.py]: This is not used.

File: credentials/urls.py [hunk 4]
```diff
@@ -39,6 +39,8 @@
     url(r'^login/$', login, name='login'),
     url(r'^logout/$', logout, name='logout'),
     url('', include('social.apps.django_app.urls', namespace='social')),
+    url(r'^credentials/', include('credentials.apps.credentials.urls', namespace='app_credentials')),
```
Comment (clintonb) [inline credentials/urls.py]: s/app_credentials/credentials

Comment (clintonb) [inline credentials/urls.py]: Alphabetize both of these lines with the others.

File: credentials/apps/credentials/urls.py [hunk 5]
```diff
@@ -1,3 +1,11 @@
 """
-URLs for the credentials service.
+URLs for the credentials views.
 """
+from django.conf.urls import url
+from credentials.apps.credentials.constants import UUID_PATTERN
+from credentials.apps.credentials.views import RenderCredential
+
+urlpatterns = [
+    url(r'^certificate/{uuid}/$'.format(uuid=UUID_PATTERN),
```
Comment (clintonb) [inline credentials/apps/credentials/urls.py]: You don't need the certificate prefix. That makes the path `/credentials/certificate/{uuid}/`. `/credentials/{uuid}/` seems sufficient.

File: credentials/apps/credentials/urls.py [hunk 6]
```diff
@@ -1,3 +1,11 @@
...
+        RenderCredential.as_view(), name='certificate_by_uuid'),
```
Comment (clintonb) [inline credentials/apps/credentials/urls.py]: s/certificate_by_uuid/render

File: credentials/apps/credentials/tests/test_views.py [hunk 7]
```diff
@@ -0,0 +1,27 @@
+"""
+Tests for credentials rendering views.
+"""
+from __future__ import unicode_literals
+from unittest import TestCase
```
Comment (clintonb) [inline credentials/apps/credentials/tests/test_views.py]: `from django.test import TestCase`

File: credentials/apps/credentials/tests/test_views.py [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+import uuid
+
+from django.core.urlresolvers import reverse
+from django.test import Client
+
+
+class RenderCredentialPageTests(TestCase):
+    """ Tests for credential rendering view. """
+
+    def setUp(self):
```
Comment (clintonb) [inline credentials/apps/credentials/tests/test_views.py]: This is unnecessary if you use Django's `TestCase`.

File: credentials/apps/credentials/tests/test_views.py [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+        super(RenderCredentialPageTests, self).setUp()
+        self.client = Client()
+
+    def test_get_valid_uuid(self):
+        """
+        Test certificate page accessibility. Page will return 200 if uuid is valid.
+        """
+        path = reverse('app_credentials:certificate_by_uuid', kwargs={
+            "certificate_uuid": uuid.uuid4().hex,
```
Comment (clintonb) [inline credentials/apps/credentials/tests/test_views.py]: Single quotes.

File: credentials/apps/credentials/constants.py [hunk 10]
```diff
@@ -3,6 +3,10 @@
 """
 from __future__ import unicode_literals
 
+# Regex used for UU-IDs.
+UUID_REGEX = r'[0-9a-f]{32}'
+UUID_PATTERN = r'(?P<certificate_uuid>{})'.format(UUID_REGEX)
```
Comment (clintonb) [inline credentials/apps/credentials/constants.py]: s/certificate_uuid/uuid

File: credentials/apps/api/tests/test_serializers.py [hunk 11]
```diff
@@ -76,7 +94,8 @@ def test_serialization_with_course_credential(self):
             ],
             "id": self.course_credential.id,
             "created": self.course_credential.created.strftime(api_settings.DATETIME_FORMAT),
-            "modified": self.course_credential.modified.strftime(api_settings.DATETIME_FORMAT)
+            "modified": self.course_credential.modified.strftime(api_settings.DATETIME_FORMAT),
+            "certificate_url": expected_url
```
Comment (clintonb) [inline credentials/apps/api/tests/test_serializers.py]: Add comma at end of line.

File: credentials/apps/api/serializers.py (hunk 1 above)
Comment (awais786) [inline credentials/apps/api/serializers.py]: This is readonly field and will be used when it returns the created credential data.

Comment (clintonb) [inline credentials/apps/api/serializers.py]: Got it.

Comment (awais786) [inline credentials/apps/api/serializers.py]: I checking it now if i can use the UserCredentialSerializer after creating the record and sending back the response.

**Comment by awais786** [comment]
@clintonb i will see the travis issue. Sorry i mentioned failed test actually build status is failed without any test failure.

**Comment by awais786** [comment]
@zubair-arbi I have made some changes in make file. Please review that.

**Comment by awais786** [comment]
@clintonb Please review this PR again. I have resolved the compression issue.

File: Makefile [hunk 12]
```diff
@@ -44,7 +44,7 @@ requirements: requirements.js
 prod-requirements:
 	pip install -qr requirements.txt --exists-action w
 
-test: clean
+test: clean static
```
Comment (clintonb) [inline Makefile]: Remove. You do not want to run static for every test run. It's slow, and unnecessary for all but the UI tests.

File: Makefile [hunk 13]
```diff
@@ -55,7 +55,7 @@ quality:
 static:
 	$(NODE_BIN)/r.js -o build.js
 	python manage.py collectstatic --noinput -v0
-	python manage.py compress -v0
+	python manage.py compress -v0 --force
```
Comment (clintonb) [inline Makefile]: You should not need to force compressor to run. You should ensure your settings are correct to allow this to run.

**Comment by clintonb** [comment]
:+1: once Makefile changes reverted.

**Comment by awais786** [comment]
@zubair-arbi please check the makefile change.

**Comment by zubair-arbi** [comment]
@awais786 I am ok with separating `clean` and `clean_static` so that the static files collected by django are not removed on `make validate` command on `travis`.

**Comment by awais786** [comment]
@clintonb I made minor change in makefile.

**Pull request merged**